### PR TITLE
Improve CloudFormation template. New development environment set up i…

### PR DESCRIPTION
**Pull request by lanami**
Problem: Eliminate present assumptions that developers have
- a pre-created AWS user set up with proper permissions to deploy Lambda functions
- a pre-created S3 Bucket
- a pre-created Lambda execution role
since this is not necessarily the case, for instance when starting from scratch on a new AWS account.
Bring deployment instructions up to date with AWS interface and terminology.

What's done:
1) Updated CloudFormation template to create an S3 bucket, a Lambda deployment user (AmazonS3FullAccess, AWSLambdaFullAccess) with an access key and a Lambda execution role.
Modified Outputs to supply developer with all necessary parameters:
- lambda deployment access key and secret
- S3 bucket name
- ARN of Lambda execution role
- app execution access key and secret

2) Migrated lamduh-maven-plugin to the most recent version (2.0.1)
3) Added to the developer's getting started guide.

### CODE REVIEW

**Comment by barbeau** [comment]
Thanks @lanami!  I'll try to review soon.

ICLA manually confirmed :+1:

File: README.md [hunk 1]
```diff
@@ -50,72 +50,92 @@ Want to make OneBusAway Alexa better?  We welcome collaboration!  See our [Contr
...
+
+Fill in the values:
+
+`skill-app-id-development` is Alexa Skill ID from previous step
+
+`aws.key-id` and `aws.secret-key` are `{appExecutionAccessKey}` and `{appExecutionSecretKey}` respectively as generated by CloudFormation
+
+`googlemaps.api-key` Google Maps GeoCoding API Key which can be obtained from [Google Developers](https://developers.google.com/maps/get-started/)
+
+`onebusaway.api-key` can be obtained from your local OBA region. Typically, you can use `TEST` while you waiting for a key.
```
Comment (barbeau) [inline README.md]: Typo - Should be "Typically, you can use `TEST` while you **are** waiting for a key.**

File: pom.xml [hunk 2]
```diff
@@ -186,14 +186,24 @@
...
-                    <functionName>onebusaway-alexa</functionName>
+                    <functionCode>
+                        ${project.build.directory}${file.separator}${project.build.finalName}-${project.packaging}-with-dependencies.jar
+                    </functionCode>
+                    <version>${project.version}</version>
+                    <lambdaFunctionsJSON>
+                        [
+                        {
+                        "functionName": "onebusaway-alexa",
+                        "description": "One Bus Away Alexa app",
```
Comment (barbeau) [inline pom.xml]: Should be "OneBusAway" - one word

Comment (barbeau) [inline pom.xml]: "app" should probably be replaced with "skill"

**Comment by lanami** [comment]
Sure, no problem. Will do some time after the 20th, sorry for the delay.

**Comment by barbeau** [comment]
Looks good,  thanks @lanami!

**Pull request merged**